openedx/core/djangoapps/programs/utils.py

```diff
@@ -211,7 +211,7 @@ def invert_programs(self):
         return inverted_programs
 
     @cached_property
-    def engaged_programs(self):
+    def engaged_programs(self) -> list[dict | None]:
```

Comment:
I did a tiny bit of type checking in this file as long as I had to learn what the possible types were during the process of figuring out what was going on in `views.py`, but I was trying to avoid scope creep in my goal, which is just improving the documentation and typing in `rest_api`.